client-app/src/admin/wip/treeMap/GridTreeMapModel.js

```diff
@@ -0,0 +1,96 @@
+import {XH, HoistModel, managed} from '@xh/hoist/core';
+import {LoadSupport} from '@xh/hoist/core/mixins';
+import {GridModel, emptyFlexCol} from '@xh/hoist/cmp/grid';
+import {DimensionChooserModel} from '@xh/hoist/desktop/cmp/dimensionchooser';
+import {TreeMapModel} from '@xh/hoist/desktop/cmp/treemap';
+import {numberRenderer, millionsRenderer, fmtNumberTooltip} from '@xh/hoist/format';
+
+@HoistModel
+@LoadSupport
+export class GridTreeMapModel {
+
+    @managed
+    dimChooserModel = new DimensionChooserModel({
+        dimensions: [
+            {value: 'region', label: 'Region'},
+            {value: 'sector', label: 'Sector'},
+            {value: 'symbol', label: 'Symbol'}
+        ],
+        initialValue: ['sector', 'symbol']
+    });
+
+    @managed
+    gridModel = new GridModel({
+        treeMode: true,
+        sortBy: 'pnl|desc|abs',
+        emptyText: 'No records found...',
+        enableColChooser: true,
+        compact: XH.appModel.useCompactGrids,
+        selModel: 'multiple',
+        columns: [
+            {
+                headerName: 'Name',
+                width: 200,
+                field: 'name',
+                isTreeColumn: true
+            },
+            {
+                field: 'mktVal',
+                headerName: 'Mkt Value (m)',
+                headerTooltip: 'Market value (in millions USD)',
+                align: 'right',
+                width: 130,
+                absSort: true,
+                agOptions: {
+                    aggFunc: 'sum'
+                },
+                tooltip: (val) => fmtNumberTooltip(val, {ledger: true}),
+                renderer: millionsRenderer({
+                    precision: 3,
+                    ledger: true
+                })
+            },
+            {
+                headerName: 'P&L',
+                field: 'pnl',
+                align: 'right',
+                width: 130,
+                absSort: true,
+                agOptions: {
+                    aggFunc: 'sum'
+                },
+                tooltip: (val) => fmtNumberTooltip(val, {ledger: true}),
+                renderer: numberRenderer({
+                    precision: 0,
+                    ledger: true,
+                    colorSpec: true
+                })
+            },
+            {...emptyFlexCol}
+        ]
+    });
+
+    @managed
+    treeMapModel = new TreeMapModel({
+        gridModel: this.gridModel,
+        labelField: 'name',
+        valueField: 'pnl',
+        heatField: 'pnl',
```

Comment:
Can we calc heat as pnl/mktVal or similar to get another dimension in the viz?  (Applies to portfolio example app as well.)